skflow/estimators/base.py

```diff
@@ -170,11 +168,12 @@ def fit(self, X, y, logdir=None):
             self._trainer.initialize(self._session)
             self._initialized = True
             # Sets up summary writer for later optional visualization
```

Comment:
the initialization to None is the hard part.
It can't be setup to None in the `__init__` because then it will be considered a param (for saving and for sklearn GridSearch stuff). Scikit learn pretty much says initialize everything you need in the `fit()`. Now because fit here is actually also partial_fit - it leads to this double logic handling if it's not there or there and `None` but just got provided with log dir.